pallets/parachain-staking/src/lib.rs

```diff
@@ -544,45 +755,448 @@ pub mod pallet {
 				})
 				.collect();
 			if let Some(balance) = amt {
-				self.nominations = OrderedSet::from(nominations);
+				self.delegations = OrderedSet::from(delegations);
 				self.total -= balance;
 				Some(self.total)
 			} else {
 				None
 			}
 		}
-		// Return false if nomination not found
-		pub fn inc_nomination(&mut self, collator: AccountId, more: Balance) -> bool {
-			for x in &mut self.nominations.0 {
-				if x.owner == collator {
-					x.amount += more;
-					self.total += more;
-					return true;
+		/// Schedule increase delegation
+		pub fn schedule_increase_delegation<T: Config>(
+			&mut self,
+			collator: AccountId,
+			more: Balance,
+		) -> Result<RoundIndex, DispatchError>
+		where
+			BalanceOf<T>: From<Balance>,
+			T::AccountId: From<AccountId>,
+		{
+			ensure!(
+				&self.delegations.0.iter().any(|x| x.owner == collator),
+				Error::<T>::DelegationDNE
+			);
+			let delegator_id: T::AccountId = self.id.clone().into();
+			ensure!(
+				T::Currency::can_reserve(&delegator_id, more.into()),
```

Comment:
should we be reserving here instead of waiting?

Reply:
No because this is scheduling the increase, not executing it. We could remove this check altogether, but I think we shouldn't allow someone to schedule the request without having the funds to execute it. They could transfer the funds out in between, but the `reserve` would still require they transfer it back in before executing.

Reply:
(if not, a test case to cover issues like "scheduled to bond more and then spent those funds before executing" could be useful)